src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6878,6 +6901,23 @@ private bool IsPossibleLocalDeclarationStatement(bool allowAnyExpression)
             }
         }
 
+        private bool IsPossibleLocalFunctionDeclaration()
+        {
+            var resetPoint = this.GetResetPoint();
+            var mods = _pool.Allocate();
+            try
+            {
+                this.ParseModifiers(mods, parsingStatementNotDeclaration: true);
```

Comment:
>this [](start = 16, length = 4)

This feels like overkill, since we only need to check if the next token is a modifier.